### PR TITLE
API Endpoints Implemented.

### PR DESCRIPTION
**Pull request by 4dbyron**
### What This PR Does
Implements API Endpoints

### Description of Task to be completed.
[chore #159222834] write tests to cover all endpoints
[feature #159222688] implement endpoint `GET /entries` : Fetch all entries
[feature #159222690] implement endpoint `GET /entries/<entryId>` : Fetch a single entry
[feature #159222726] implement endpoint `POST /entries` : Create an entry
[feature #159222752] implement endpoint `PUT /entries/<entryId>` : Modify an entry
[feature #159331101] implement endpoint `DELETE /entries/<entryId>` : Deletes the specified entry.
[chore #159222903] test each newly implemented endpoint with postman

#### 1. Clone the repository with: 
`git clone https://github.com/4dbyron/MyDiary.git`
or
`git clone git@github.com:4dbyron/MyDiary.git`

### To Test using Postman
- start the server with `python app.py`
- launch Postman and enter in the desired endpoint to test.
Example ```GET /entries``` fetch all entries.

## Relevant Pivotal Tracker Stories

#159222834
#159222690
#159222690
#159222726
#159222752
#159331101
#159222903
[Take Me To Those Stories](https://www.pivotaltracker.com/n/projects/2184264)

### CODE REVIEW

File: api/test_app.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        response = self.my_diary.get(self.path + '/entries')
+        self.assertEqual(response.status_code, 200)
+
+    def test_receive_specific_entry(self):
+        """return specific diary entry"""
+        _id = {"entry_id": 3}
+        response = self.my_diary.get(self.path + '/entries/{}'.format(_id['entry_id']))
+        self.assertEqual(response.status_code, 200)
+
+    def test_post_new_entry(self):
```
Comment (NgangaMaryanne) [inline api/test_app.py:30]: The POST  method should be done to the url: `/api/v1/entries` not `/api/v1/entries/<int:entry_id>` . A user should not have to pass in an `id` to the POST. The id should be added automatically.

File: api/test_app.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+class TestEndPoints(unittest.TestCase):
+    """Set Defaults"""
+
+    def setUp(self):
+        self.my_diary = app.app.test_client()
+        self.path = '/api/v1'
+
+    # The 4 main tests
+    def test_receive_entries(self):
```
Comment (NgangaMaryanne) [inline api/test_app.py:19]: Its good practice to name tests after the name of the function being tested or at least an implication of it. ie. This test would have been better if named ` def test_get_entries(self)`

File: api/test_app.py (hunk 1 above)
Comment (4dbyron) [reply api/test_app.py:30]: I will fix it before progressing on. Thank You For the Feedback

File: api/test_app.py (hunk 2 above)
Comment (4dbyron) [reply api/test_app.py:19]: All the mentioned issues are now being fixed thanks to Your Feedback.
Thank You For Your Support. :smiley:

**Pull request merged**